pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -749,27 +785,46 @@ func (ku *KubeUtil) setKubeletHost(hosts *connectionInfo, httpsPort, httpPort in
 func selectFromPotentialHostsHTTPS(ku *KubeUtil, hosts []string, httpsPort int) (string, []error) {
 	var connectionErrors []error
 	for _, host := range hosts {
+		verifyTLS := config.Datadog.GetBool("kubelet_tls_verify")
+
 		log.Debugf("Trying to use host %s with HTTPS", host)
-		ku.kubeletHost = host
-		err := ku.setupkubeletAPIClient()
+		err := tryKubeletConnection(ku, host, httpsPort, verifyTLS)
 		if err != nil {
-			log.Debugf("Cannot setup https kubelet api client for %s: %v", host, err)
 			connectionErrors = append(connectionErrors, err)
-			continue
-		}
-
-		err = checkKubeletHTTPSConnection(ku, httpsPort)
-		if err == nil {
-			log.Debugf("Can connect to kubelet using %s and HTTPS", host)
-			return host, nil
+			if config.Datadog.GetBool("kubelet_fallback_to_unverified_tls") && verifyTLS {
+				unverifiedTLSAttemptWarning(err)
+				err = tryKubeletConnection(ku, host, httpsPort, false)
+				if err != nil {
+					connectionErrors = append(connectionErrors, err)
+					continue
+				}
+			} else {
+				continue
+			}
 		}
-		log.Debugf("Cannot connect to kubelet using %s and https: %v", host, err)
-		connectionErrors = append(connectionErrors, err)
+		return host, nil
 	}
 
 	return "", connectionErrors
 }
 
+func tryKubeletConnection(ku *KubeUtil, host string, httpsPort int, verifyTLS bool) error {
+	ku.kubeletHost = host
+	err := ku.setupKubeletAPIClient(verifyTLS)
+	if err != nil {
+		log.Debugf("Cannot setup https kubelet api client for %s: %v. TLS verification: %v.", host, err, verifyTLS)
+		return err
+	}
+
+	err = checkKubeletHTTPSConnection(ku, httpsPort)
+	if err == nil {
+		log.Debugf("Can connect to kubelet using %s and HTTPS. TLS verification: %v.", host, verifyTLS)
```

Comment:
```suggestion
		log.Debugf("Can connect to kubelet using %s:%d and HTTPS. TLS verification: %v.", host, httpsPort, verifyTLS)
```

Reply:
i'd use some more positive sentence)) like "Connection to kubelet is successful"